### PR TITLE
Ft all info for exercise 149466925

### PR DESCRIPTION
**Pull request by collin5**
## PR Status
**READY**

## Story ID
[#149466925] Show all infos for an exercise

## Added Migrations
NO

## Description
Extend API to show all info of exercise with `?detail=all` query string modification 
ie 
```
    http://127.0.0.1:8000/api/v2/exercise?detail=all
```
For information about a single exercise item `?id=<id>` use combination `?id=<id>&detail=all` for more detailed information
ie
```
http://127.0.0.1:8000/api/v2/exercise?id=345&detail=all
```

## New Packages Added
None

## Todos
- [x] Tested and working locally
- [x] Tested and working on development environment

## Impacted Areas in Application
List general components of the application that this PR will affect:
*  wger/exercises/api/serializers.py
*  wger/exercises/api/serializers.py

### CODE REVIEW

**Comment by johnseremba** [comment]
@collin5 Thanks for the good work.
I have looked at the api and you seem to be displaying data for all exercises. However the story description from my perspective requires you to display all information for a particular exercise.
The story states `Show all infos for an exercise`, but not `Show all infos for all exercises`. 
I think you need to check that.

File: wger/exercises/api/serializers.py [hunk 1]
```diff
@@ -71,3 +72,22 @@ class MuscleSerializer(serializers.ModelSerializer):
...
+    category = ExerciseCategorySerializer()
+    license = LicenseSerializer()
+    language = LanguageSerializer()
+
+    def get_exercise_images(self, obj):
+        img_query = ExerciseImage.objects.filter(exercise_id=obj.id).all()
+        return ExerciseSerializer(instance=img_query, many=True).data
+
+    class Meta:
+        model = Exercise
```
Comment (johnseremba) [inline wger/exercises/api/serializers.py:94]: I think you need to query for a particular exercise.

File: wger/exercises/api/views.py [hunk 2]
```diff
@@ -54,17 +72,13 @@ class ExerciseViewSet(viewsets.ModelViewSet):
...
-                     'status',
-                     'name',
-                     'equipment',
-                     'license',
-                     'license_author')
+    filter_fields = Shared.exercise_filter_fileds()
+
+    def get_queryset(self):
+        # if ?detail=all in query string
+        if self.request.GET.get('detail') == 'all':
```
Comment (johnseremba) [inline wger/exercises/api/views.py:80]: You may need to pass in the id of an exercise instead.

File: wger/exercises/api/serializers.py [hunk 3]
```diff
@@ -71,3 +72,22 @@ class MuscleSerializer(serializers.ModelSerializer):
...
     class Meta:
         model = Muscle
+
+
+class DetailedExerciseSerializer(serializers.ModelSerializer):
+    '''
+    Exercises with detail
+    '''
+    equipment = EquipmentSerializer(many=True)
+    muscles, muscles_secondary = MuscleSerializer(many=True), MuscleSerializer(many=True)
```
Comment (johnseremba) [inline wger/exercises/api/serializers.py]: Flake8 will definitely complain. Line is too long.

File: wger/exercises/api/serializers.py (hunk 1 above)
Comment (collin5) [reply wger/exercises/api/serializers.py:94]: Fixed, supports query string attr `id=<id>` now

File: wger/exercises/api/views.py (hunk 2 above)
Comment (collin5) [reply wger/exercises/api/views.py:80]: Solved

File: wger/exercises/api/serializers.py (hunk 3 above)
Comment (collin5) [reply wger/exercises/api/serializers.py]: Changed line length to 79, solved

**Review by johnseremba — APPROVED** [review]
I seems good to me.

**Pull request merged**